Combinatorial_map/include/CGAL/Combinatorial_map.h

```diff
@@ -50,6 +50,11 @@
 #include <deque>
 #include <boost/type_traits/is_same.hpp>
 #include <boost/unordered_map.hpp>
+#include <boost/graph/graph_traits.hpp>
+#include <CGAL/boost/graph/helpers.h>
+#include <CGAL/boost/graph/graph_traits_Polyhedron_3.h>
+#include <CGAL/boost/graph/graph_traits_Surface_mesh.h>
```

Comment:
This file should be include anything specific to `Polyhedron_3` or `Surface_mesh`. That is why the dependencies increased that much, including bad dependencies to GPL headers.

Reply:
I don't see why you would need to include those graph_traits files anyway.

Reply:
For the method "import_from_halfedge_graph", which takes an halfedge graph as input and build a CMap.  Without these include I have error, for example 

‘next’ is not a member of ‘CGAL’
                         (*origin_to_copy)[CGAL::next(dartmap_iter->first, heg)],

Reply:
I can remove the #include <CGAL/boost/graph/graph_traits_Surface_mesh.h>, keeping the others. It should be ok since BGL and Polyhedron are both LGPL.

Reply:
The functions described in the BGL concepts must be called by [ADL], not qualified by `CGAL::`. Otherwise the code cannot compile on a non-CGAL face graph where the `next` function can be in the `openmesh` namespace, for example.

[ADL]: https://en.cppreference.com/w/cpp/language/adl

Reply:
I am not able to remove the CGAL::
Indeed, in the CMap class, I have a method next(Dart_handle). Without CGAL::next the compiler complains that there is no next function with two parameters.

@lrineau Do you have any solution to this problem ?

Reply:
Having a member function named `next` blokes the ADL lookup. I am looking for a solution...

Reply:
One possibility is to say that my method that converts a model of face graph into a combinatorial maps works only for CGAL models. Then I can keep the CGAL::

Reply:
`Surface_mesh` also has a `next` member function and I'm not aware of any issue due to that. The issue should be somewhere else?

Reply:
Does a method of Surface_mesh calls a next of bgl? Probably not.

Reply:
A member function or a free function ?

Reply:
A member function of Surface_mesh which calls a free function next ?

Reply:
In Surface_mesh, there is nowhere where `CGAL::next` is called on a face graph type that is not `Surface_mesh` itself. So probably the calls are qualified, with `CGAL::`.

Reply:
Ah I understand now. Making your import function a free function should solve the issue. I don't have time now but I'm willing to help if needed.

Reply:
I have a small patch in hand. I am testing it fully...

Reply:
I have pushed a small fix in 7959a40e68f. Guillaume, do you agree with the patch?

In passing, I have added a test of the function `import_from_halfedge_graph` with an alien face graph (from OpenMesh). CTest tests pass on my machine, and I have verified that `import_from_halfedge_graph` does not compile with OpenMesh if the patch to `Combinatorial_map/include/CGAL/Combinatorial_map.h` is reverted. So the fix does fix something.

I have not fixed the `dependencies` files. For the moment they are wrong no matter what. Let's wait for Travis to tell us what to modify.

Reply:
Thanks @lrineau  !
About your patch, I suppose I agree if it works both for Surface_mesh and OpenMesh. 
Thanks again !